### PR TITLE
Allowing hosts to share the mounted volumes as either read only or as SELinux write compatiable volume.

### PR DESCRIPTION
**Pull request by ssvaddiparthy**
## Changelog

Fill this in and update CHANGELOG.md

## Questions (open the PR then click the check boxes)

Did you update the documentation related to your changes?

- [x] Yes
- [ ] My changes were not already documented

Did you run `make` _before_ committing code and opening this PR?

- [x] Yes
- [ ] I didn't change any code

Did you run `porter create-stack` and `porter sync-stack` to verify provisioning
works?

- [x] Yes
- [ ] N/A

### CODE REVIEW

File: hook/hook.go [hunk 1]
```diff
@@ -240,10 +240,27 @@ func ExecuteWithRunCapture(log log15.Logger,
...
+	var mountedVolume, volumeFlag string
+	mountedVolume = "/repo_root"
+	volumeFlag = os.Getenv(constants.EnvVolumeFlag)
+
+	// Adding volume flag to either allow read only or to maintain same volume flags
+	switch volumeFlag {
+	case "z", "ro":
+	default:
+		log.Info("Setting the default volume flag as empty.")
+		volumeFlag = ""
```
Comment (phylake) [inline hook/hook.go]: This is the wrong validation. If the value was set we need to ensure it's one of the valid values or reset it to `""`. This should read

```golang
case "", "z", "ro":
default:
    log.Warn("invalid volume flag set. resetting to \"\"", "volumeFlag", volumeFlag)
    volumeFlag = ""
```

alternatively we change the signature of this function to return `(runArgs []string, success bool)`

File: commands/help/debug.go [hunk 2]
```diff
@@ -73,7 +73,15 @@ STACK_CREATION_ON_FAILURE
...
-    Grease the wheels on development`
+
+    Grease the wheels on development
+
+VOLUME_FLAG
+    
+    By default the repo's root is volume mapped to /repo_root with no volume mounting options.
+    This environment variable now allows the volume mount to be either shared as:
+        a. Read only when VOLUME_FLAG is set to 'ro'.
+        b. SELinux write compatible with labels preserved when VOLUME_FLAG is set to 'z' `
```
Comment (phylake) [inline commands/help/debug.go]: you have an extra space here `'z' `

File: hook/hook.go [hunk 3]
```diff
@@ -240,10 +240,26 @@ func ExecuteWithRunCapture(log log15.Logger,
...
+	var mountedVolume, volumeFlag string
+	mountedVolume = "/repo_root"
+	volumeFlag = os.Getenv(constants.EnvVolumeFlag)
+
+	// Adding volume flag to either allow read only or to maintain same volume flags
+	switch volumeFlag {
+	case "", "z", "ro":
+	default:
+    	log.Warn("invalid volume flag set. resetting to \"\"", "volumeFlag", volumeFlag)
+    	volumeFlag = ""
```
Comment (phylake) [inline hook/hook.go]: Looks like you need to run `make` so formatting is run

**Comment by phylake** [comment]
Looks good just run `make` and commit any changes

**Comment by ssvaddiparthy** [comment]
I have added the formatting changes.

**Comment by phylake** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/adobe-platform/porter/pull-request/183/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=adobe-platform/porter)

**Comment by phylake** [comment]
@davepersing can you take a look?

**Comment by davepersing** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/adobe-platform/porter/pull-request/183/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=adobe-platform/porter)

**Pull request merged**